pkg/k8s/k8s_service.go

```diff
@@ -142,3 +144,92 @@ func prepareService(
 	}
 	return svc, nil
 }
+
+func (c *Client) WaitForService(ctx context.Context, name string) error {
```

Comment:
This should have a unit test if possible. If it did I suspect my comments on the for loop not blocking would have been caught.

Reply:
that's definitely doable. Let's handle it in a seprate PR.